### PR TITLE
Update documentation and examples

### PR DESCRIPTION
**Pull request by andreeaflorescu**
These updates are preparing the vm-fdt crate to be published on crates.io.

The following changes are part of this PR:
- update the README.md with a few more details about the design
- add high level examples of using the interface in the README.md as well as in src/lib.rs
- add separate examples for other functions that are not already called into the main example

Fixes https://github.com/rust-vmm/vm-fdt/issues/15

### CODE REVIEW

**Comment by andreeaflorescu** [comment]
For these examples to work we need to cherry-pick the commit from #21. The code examples thus serve as a test that we're exporting the right types as well.

File: README.md [hunk 1]
```diff
@@ -3,47 +3,90 @@
...
-- Does the crate have any default/optional features? What is each feature
-  doing?
-- Is this crate used by other rust-vmm components? If yes, how?
+## Usage
 
-This crate defines a development feature: `long_running_test`. This feature
-SHOULD NOT be used in production as it might enable functionality that is safe
-only for development use cases.
+The following code is creating an FDT blob with a root node that has 3
+properties (“compatible”, “#address-cells”, and “#size-cells”), and contains
```
Comment (danielverkamp) [inline README.md:49]: This text has some fancy quotes in it (“” vs "") - not sure if that matters (e.g. if somebody tries to copy and paste the strings into code.

File: README.md [hunk 2]
```diff
@@ -3,47 +3,90 @@
...
+2 child nodes:
+1.  “chosen”, which has 2 properties (“linux,pci-probe-only” and “bootargs”).\
```
Comment (danielverkamp) [inline README.md:51]: The first numbered list item has two spaces after it whereas the next line only has one.

There is also a `\` at the end of the first item, which is probably not intended.

**Review by danielverkamp — CHANGES_REQUESTED** [review]
Thanks, this looks good to me overall.

File: README.md (hunk 2 above)
Comment (andreeaflorescu) [reply README.md]: Fixed.

File: README.md (hunk 1 above)
Comment (andreeaflorescu) [reply README.md]: Fixed.

**Comment by andreeaflorescu** [comment]
I also ran the markdown linter and fixed a few other things (some too long lines, spaces needed before and after list, ordered list prefix to be 1). The full list is available here: https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md

**Review by danielverkamp — APPROVED** [review]
Thanks!

**Review by alexandruag — APPROVED** [review]

**Pull request merged**